src/shared/logs-show.c

```diff
@@ -333,8 +334,40 @@ static int output_timestamp_monotonic(FILE *f, sd_journal *j, const char *monoto
         if (r < 0)
                 return log_error_errno(r, "Failed to get monotonic timestamp: %m");
 
-        fprintf(f, "[%5"PRI_USEC".%06"PRI_USEC"]", t / USEC_PER_SEC, t % USEC_PER_SEC);
-        return 1 + 5 + 1 + 6 + 1;
+        snprintf(buf, sizeof(buf), "[%5"PRI_USEC".%06"PRI_USEC"", t / USEC_PER_SEC, t % USEC_PER_SEC);
+
+        if (mode == OUTPUT_SHORT_DELTA) {
+                sd_id128_t previous_boot_id;
+                uint64_t previous_t, delta;
+
+                r = sd_journal_previous(j);
+                if (r <= 0)
+                        delta = 0;
+                else {
+                        r = sd_journal_get_monotonic_usec(j, &previous_t, &previous_boot_id);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to get monotonic timestamp: %m");
+
+                        r = sd_journal_next(j);
+                        /* This should actually always succeed because we are coming from this state but to be sure ... */
+                        if (r < 0)
+                                return r;
+                        else if (r == 0)
+                                return log_debug_errno(SYNTHETIC_ERRNO(ENODATA), "Whoopsie! We cannot go back to our previously used journal entry.");
+
+                        if (sd_id128_equal(boot_id, previous_boot_id))
+                                delta = usec_sub_unsigned(t, previous_t);
+                        else
+                                delta = 0;
```

Comment:
i'd prefer if we'd use realtime timestamps as fallback here. i.e. keep track of boot id + dual_timestamp of the previously printed line and then use monotonic time if possible and realtime otherwise.

Reply:
Done. I've marked those deltas with a `*`. For example

```
[    5.858059 <   19.580722*>] zeus systemd[1]: Starting Network Configuration...
```